ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/reaper/ReaperTestCase3.java

```diff
@@ -229,8 +229,15 @@ public void testReaperForce() throws Exception
         // byteman rules will ensure that the reaper and reaperworker rendezvous gte deleted
         // under this call
 
+        long start = System.currentTimeMillis();
```

Comment:
Given the comment `force a termination and don't wait for the normal timeout periods` I personally would be fine with your additional check of how long the timeout takes and making sure it's shorter than the timeout in this test but please can you clarify why you needed to modify the timeouts of the inserts above? Hopefully reapable3 being 4 seconds would generally be long enough on it's own to be able to show that the reaper was terminated in a time less than the timeout of a Reapable? We might consider allowing just over a second for the theoretical scenario where the wait(1000) takes the full length of time.

Reply:
> but please can you clarify why you needed to modify the timeouts of the inserts above?

Because that change was made initially to highlight the issue.

As far as what the could or should be, I don't have too strong of an opinion. There are two things that need accounted for - that reaping begins immediately after shutdown and does not wait for the first timeout (this is the effect of setting  nextDynamicCheckTime to 0 and notifying all), and that you do not wait any of the remaining transactions to fully timeout.

Reply:
We should have the test removed from the exclusion in https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/pom.xml#L61 (thanks @jmfinelli for the guidance) so builds and CI runs it /cc @mmusgrov

Reply:
Changed the timeouts back.

Reply:
Thank you!